### DISCUSSION TITLE
Noob question: creating a package

### DISCUSSION THREAD

**Discussion by lveillard** [vercel/turborepo#2057] [Help] [answered]
Hello all!
I'm creating a package/tools following the same structure as https://github.com/vercel/turborepo/tree/main/packages/turbo-utils

It did not work directly by having a index.ts so i tried to replicate exactly the same thing that the turborepo monorepo has with tsup.

However, even after including my package in the dependencies on the package.json file, it does not work: `Can't resolve 'test-outils'`

I realized there is no 'dev' script in turbo-utils, only a build script. Migth this be the problem that I can't access my `/packages/test-outils` package?

I can properly run check-types, build and lint using pnpm  --filter test-outils and i've `run pnpm i` but still no way to import stuff from my package

Would love to get some help if anyone has a working example.

Also, a related question: Why using tsup to load /dist/index.ts  and not directly importing /src/index.ts?

Stuff i've done:
1) I'm importing it properly in the json.package like this:
```
 "dependencies": {
    "test-tools": "workspace:*"
  }
```

2) I've configured 'test-outils' like this:
```
...
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "tsup",
    "lint": "eslint src/**/*.ts",
    "types": "tsc --noEmit"
  },
```

Everything like the turborepo monorepo. What i am missing?

Thanks!

**Comment by lveillard** [comment, marked as answer]
So for anyone reading this... yep obviously the dev command requires also to run "tsup".

Still not sure how turborepo devs work in dev as they don't have a dev command and if anyone knows why do they use tsup in the first place, I would love to know! 

```
  "scripts": {
    "dev": "tsup",
    ...
```

**Comment by anthonyshew** [comment]
You'll likely find [the examples folder](https://github.com/vercel/turbo/tree/main/examples) to be much clearer for learning purposes.

I'm not quite grokking the heart of your question but worthy of note: tsup has a `--watch` flag.
